### ISSUE TITLE
Move from Glide to Dep: dep ensure -add fails

### ISSUE THREAD

**Issue by lawrencegripper** [Azure/azure-sdk-for-go#1108]
Currently docs suggest the following method to import the SDK with `dep` however this fails with the latest version of dep 

```
dep:
 version     : v0.4.1
 build date  : 2018-01-27
 git hash    : 37d9ea0
 go version  : go1.9.3
 go compiler : gc
 platform    : darwin/amd64
``` 

I created a new package with a `main.go` then ran `dep init` then tried to import the sdk:

 ```
~/ dep ensure -add github.com/Azure/azure-sdk-for-go@14.1.1
Fetching sources...

Solving failure: No versions of github.com/Azure/azure-sdk-for-go met constraints:
	14.1.1: Could not introduce github.com/Azure/azure-sdk-for-go@14.1.1, as its subpackage github.com/Azure/azure-sdk-for-go does not contain usable Go code (*build.NoGoError).. (Package is required by (root).)
	v14.1.1: Could not introduce github.com/Azure/azure-sdk-for-go@v14.1.1, as its subpackage github.com/Azure/azure-sdk-for-go does not contain usable Go code (*build.NoGoError).. (Package is required by (root).)
	v14.1.0: Could not introduce github.com/Azure/azure-sdk-for-go@v14.1.0, as it is not allowed by constraint ^14.1.1 from project github.com/lawrencegripper/azuresdkgotest.
	v14.0.2: Could not introduce github.com/Azure/azure-sdk-for-go@v14.0.2, as it is not allowed by constraint ^14.1.1 from project github.com/lawrencegripper/azuresdkgotest.
.....
```

It seems that adding a sub-directory succeeds. 

``` 
✘ dep ensure -add github.com/Azure/azure-sdk-for-go/services/batch/2017-09-01.6.0/batch@14.1.1
Fetching sources...

"github.com/Azure/azure-sdk-for-go/services/batch/2017-09-01.6.0/batch" is not imported by your project, and has been temporarily added to Gopkg.lock and vendor/.
If you run "dep ensure" again before actually importing it, it will disappear from Gopkg.lock and vendor/.
```

This creates a gopkg.lock like so 

```toml
[[projects]]
  name = "github.com/Azure/azure-sdk-for-go"
  packages = [
    "services/batch/2017-09-01.6.0/batch",
    "version"
  ]
  revision = "1e334c402ea1460704b0263e5d188f28ad946ce1"
  version = "14.1.1"
```

I can then go into my go code and start using it, at which point the warning isn't a problem any more. 

It looks like this is being worked on by dep [here](https://github.com/golang/dep/pull/1545) and [here](https://github.com/golang/dep/issues/1306)

I'm relatively new to `dep` having spent some time getting my head around `glide` so this may simply be user error. If it's not then I think it would be good to update the docs to call this out. 

Any help and pointers much appreciated!

**Comment by joshgav** [comment]
Thanks for noting this @lawrencegripper. I think the solution may be to turn the root into a Go package by adding a `doc.go` file. Did so in #1310.

**Comment by lawrencegripper** [comment]
Awesome that sounds like it will do the trick - good call

**Comment by vladbarosan** [comment]
Fixed by #1310
